src/main/groovy/edu/oregonstate/mist/textbooksapi/TextbooksCollector.groovy

```diff
@@ -163,24 +147,53 @@ class TextbooksCollector {
         }
         res
     }
+
+    Response validateTerm(String academicYear, String term) {
+        String termString = "${academicYear}-${term}"
+        def res = getResponse(coursesUri)
+        List<Term> termList = getListFromResponse(res, Term.class)
+        if(termList.find { it.id == termString }) {
+            null
+        } else {
+            Resource.badRequest("Invalid term: '${termString}'. Valid terms are [" +
```

Comment:
I would advise against one of your DAO methods returning a `Response` object. We want to have some separation of concerns between your resource class and your DAO/textbookcollector class. For example, your DAO class should abstract the fact that a term is actually the year and term concatenated.

Regarding validating the term, you could have this method return a map where the key is the year and the value is an array of terms. For example `def validTerms = [2018: ["fall", "summer"]]` or `def validTerms = [2018:["fall"], 2019 ["winter"]]`

Then your resource would check to see if the term and year the client provided are valid, given the contents of that map.

Another option would be to have your resource class just send all the parameters to the DAO method. Then your DAO method could validate the term and year, and if they were invalid, it could throw a new `TextbookCollectorException` (for example). Then your resource class could catch that exception and return a badRequest, since your exception would mean the year or term was invalid.